### DISCUSSION TITLE
Dynamic deep links to filter taxonomy field

### DISCUSSION THREAD

**Discussion by BrinaD1** [microsoft-search/pnp-modern-search#1962] [Q&A]
Hello, I am setting up a news rollup page using the search web parts and we are tagging the news with a MMD field. I've customized a display template that has the tags as links at the bottom with the idea that you click on the tag and it will filter the page to all articles with that tag. 
![image](https://user-images.githubusercontent.com/5232243/160180223-0f895894-0cbb-4e81-b927-c70ce1e49235.png)

My refiner is RefinableString174 mapped to the taxonomy field's crawled property ows_taxid_Tags. I've hacked together some crazy handlebars code to actually split out the term GUID and name and create a URL from it for the user to click. As I was testing, though, it never worked. I realized that if you go to the page and filter the Tags filter on the page, the URL there is different from what I was putting together. It doesn't seem to be the term GUID at all but some other value. 

Search Page Filtered to specific tag - Decoded 
https://tenant.sharepoint.com/sites/net/SitePages/Search-Test-Tags.aspx?f=[{"filterName":"RefinableString174","values":[{"name":"Education","value":"\"ǂǂ4750307c2337666338393138332d656361352d343035622d396364332d363263376165366163363162\"","operator":0}],"operator":"or"}]#

Our generated URL - Decoded
https://tenant.sharepoint.com/sites/net/SitePages/Search-Test-Tags.aspx?f=[{"filterName":"RefinableString174","values":[{"name":"Education","value":"\"ǂǂ7fc89183-eca5-405b-9cd3-62c7ae6ac61b\"","operator":0}],"operator":"or"}]#

![image](https://user-images.githubusercontent.com/5232243/160695041-90956ba9-d863-4e04-b2b1-1fbf34e8c9bc.png)

Our generated URL will filter the web part but we get no results.
![image](https://user-images.githubusercontent.com/5232243/160181730-256dcd21-2a30-41f9-a509-aaf78690fe9e.png)

What is the term value being used in the URL when you get the url from filtering the filters web part? How is that being generated and why does it not match the term GUID? The Tags term set is open and is folksonomy with over 200 terms so it doesn't make sense to try to make a mapping of all of them and hard code. 

Thanks!

**Comment by BrinaD1** [comment]
I'm trying to follow up on this. I'd like to dynamically create the links that were removed in the issue #1729. I've written the handlebars code to parse my tags and create links but they don't work. What is happening to the term ID when you use the filter web part on the page and it constructs the Url? Is there an existing function I could call in the handlebars layout to do the same thing? Thanks!

**Reply by FranckyC** [reply]
Hi @BrinaD1, the links you mention from #1729 were simply anchors with the term label as value. It was meant to be used by an other search results WebPart with 'Query fragment' configured as input source.  It was different from filter deep links.

Filter deep links are actually "refinement filters". The reason you don't see anything is because you don't have a base query to retrieve a set of results first meaning you can't refine on an empty result set basically. To get it work, you can build your links with a `q=*` query string parameter (or a query to retrieve only news) and configure the search results Web Part to use that parameter as default 

You can also configure the query template with the same base query so you get results at page load and then refine them afterwards.

Finally to answer your questions:

> What is the term value being used in the URL when you get the url from filtering the filters web part? How is that being generated and why does it not match the term GUID?

It is the `ǂǂ` + HEX encoded value of the term GUID representation. See by yourself https://www.convertstring.com/en/EncodeDecode/HexDecode.

**Reply by BrinaD1** [reply]
Thanks for the info! I do have a base query that loads all the news items to start  and then the user clicks a tag on the article to refine it down as opposed to opening up the refinement panel and death scrolling until they find the tag they want. Some of the tags are similar so I didn't want to paste the tag into the query box but wanted to have it actually filter down using the refinement panel. 

I have it all working except for the HEX encoding. I'm assuming there is no handlebars helper that does that OOTB but we would have to include an extension?
